allennlp/common/from_params.py

```diff
@@ -152,11 +152,15 @@ def infer_params(cls: Type[T], constructor: Callable[..., T] = None):
         return parameters
 
     # "mro" is "method resolution order".  The first one is the current class, the next is the
-    # first superclass, and so on.  Taking the first superclass should work in almost all cases that
-    # we're looking for here.  This could fail, though, if you are using multiple inheritance and we
-    # pick the wrong superclass.  We'll worry about how to fix that when we run into an actual
-    # problem because of it.
-    super_class = cls.mro()[1]
+    # first superclass, and so on.  We take the first superclass we find that inherits from
+    # FromParams.
+    super_class = None
+    for super_class_candidate in cls.mro()[1:]:
+        if issubclass(super_class_candidate, FromParams):
+            super_class = super_class_candidate
+            break
+    if not super_class:
```

Comment:
I prefer `if super_class is None:`. There are many things in Python that are falsy. It can be confusing.

Of course, that style is common in the library, and one more instance doesn't make much of a difference.

Reply:
Yeah, I do that in a bunch of cases, where there's a chance for error.  I don't think it's needed here - classes returned by `mro()` are not going to be falsy.  I'll also note that `if not X` is not just common in the library, it's recommended, idiomatic python usage.